Tests/Brokerages/OrderCrossingBrokerageTests.cs

```diff
@@ -184,20 +175,100 @@ public void PlaceCrossOrderAndUpdate(Order leanOrder, OrderStatus[] expectedOrde
-            var updateSubmittedOrder = brokerage.GetAllOrders(o => o.Status == OrderStatus.UpdateSubmitted).Single();
-            Assert.IsNotNull(updateSubmittedOrder);
+            AssertComingOrderStatusByEvent(autoResetEventUpdateSubmittedStatus, brokerage, OrderStatus.UpdateSubmitted);
+
+            AssertComingOrderStatusByEvent(autoResetEventFilledStatus, brokerage, OrderStatus.Filled);
+
+            CollectionAssert.AreEquivalent(expectedOrderStatusChangedOrdering, actualCrossZeroOrderStatusOrdering);
+            Assert.AreEqual(0, brokerage.GetLeanOrderByZeroCrossBrokerageOrderIdCount());
+        }
+
+        private static IEnumerable<TestCaseData> CrossZeroInvalidFirstPartParameters
+        {
+            get
+            {
+                var stopMarketOrder = new StopMarketOrder(Symbols.AAPL, -20, 180m, DateTime.UtcNow);
+                yield return new TestCaseData(stopMarketOrder, new[] { OrderStatus.Submitted, OrderStatus.Invalid });
+            }
+        }
+
+        [Test, TestCaseSource(nameof(CrossZeroInvalidFirstPartParameters))]
+        public void PlaceCrossOrderInvalid(Order leanOrder, OrderStatus[] expectedOrderStatusChangedOrdering)
+        {
+            var actualCrossZeroOrderStatusOrdering = new Queue<OrderStatus>();
+            using var autoResetEventInvalidStatus = new AutoResetEvent(false);
+
+            using var brokerage = InitializeBrokerage((leanOrder?.Symbol.Value, 180m, 10));
+
+            brokerage.OrdersStatusChanged += (_, orderEvents) =>
+            {
+                var orderEventStatus = orderEvents[0].Status;
+
+                actualCrossZeroOrderStatusOrdering.Enqueue(orderEventStatus);
+
+                Log.Trace($"{nameof(PlaceCrossOrder)}.OrdersStatusChangedEvent.Status: {orderEventStatus}");
+
+                if (orderEventStatus == OrderStatus.Invalid)
+                {
+                    autoResetEventInvalidStatus.Set();
+                }
+            };
+
+            brokerage.IsPlaceOrderPhonyBrokerageFirstPartSuccessfully = false;
+            brokerage.IsPlaceOrderPhonyBrokerageSecondPartSuccessfully = false;
+
+            var response = brokerage.PlaceOrder(leanOrder);
+            Assert.IsFalse(response);
+
+            AssertComingOrderStatusByEvent(autoResetEventInvalidStatus, brokerage, OrderStatus.Invalid);
+
+            CollectionAssert.AreEquivalent(expectedOrderStatusChangedOrdering, actualCrossZeroOrderStatusOrdering);
+            Assert.AreEqual(0, brokerage.GetLeanOrderByZeroCrossBrokerageOrderIdCount());
+        }
+
+        private static IEnumerable<TestCaseData> OrderCrossZeroSecondPartParameters
+        {
+            get
+            {
+                var stopMarketOrder = new StopMarketOrder(Symbols.AAPL, -20, 180m, DateTime.UtcNow);
+                yield return new TestCaseData(stopMarketOrder, new[] { OrderStatus.Submitted, OrderStatus.PartiallyFilled, OrderStatus.Filled, OrderStatus.Canceled });
```

Comment:
Add expected OrderStatus for order events from the perspective of algorithm, in this case it should be `OrderStatus.Submitted, OrderStatus.PartiallyFilled, OrderStatus.Canceled`